pages/profile/[username].tsx

```diff
@@ -82,9 +82,9 @@ const UserProfile: React.FC = () => {
     const lesson = lessonInfo || {}
     const challengeList = lesson.challenges || []
     const challengesStatus = challengeList.map(challengeInfo => {
-      const challenge = challengeInfo || {}
+      const challenge = challengeInfo || ({} as Challenge)
       const challengeSubmission = userSubmissions.find(
-        submission => challenge.id === submission.challengeId
+        submission => +challenge.id === +submission.challengeId
```

Comment:
Avoid using implicit conversion. Having an explicit conversion makes the code clearer as that little `+` might go unnoticed.

Since `challenge.id` is already `number`, you only need to convert `submission.challengeId`.

```suggestion
        submission => challenge.id === Number(submission.challengeId)
```

Reply:
Ah, VSCode was slow in updating the Challenge type so it still showed Maybe string and flagged it. Thanks.